core/order.go

```diff
@@ -821,11 +883,14 @@ func (n *OpenBazaarNode) CalculateOrderTotal(contract *pb.RicardianContract) (ui
 				}
 			}
 		}
-		itemTotal *= uint64(item.Quantity)
+		itemTotal *= uint64(itemQuantity)
 		total += itemTotal
 	}
 
-	// Add in shipping costs
+	// Add in shipping costs for physical items
+	if len(physicalGoods) == 0 {
+		return total, nil
+	}
```

Comment:
Instead of short circuiting the whole function, could be just protect the shipping calculations inside an IF block? I don't want to have to be aware that non-physical goods won't get processed behind this point.

Reply:
Or even better, extract this behavior into a function which returns the appropriate shipping value to be attached to the order?